### PR TITLE
Catch undefined cases in simplified login

### LINKED ISSUE #1881: Cannot log in with simplified log in for usernames over 3 characters
## System information

*Please specify the Kolibri version you were using and your operating system!*

 - Version: Kolibri 0.5.0beta4.dev22
 - Operating system: MacOS Sierra
 - Browser: Chrome

## How to reproduce

1. Activate simplified login 
2. Log out
3. Log in with a username that is more than 3 characters long

## Screenshots

screen shot 2017-07-20 at 3 01 43 pm

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Be more wary of emitted data and check for nulls and undefineds in the simplified login logic.

Hopefully fixes #1881

### CODE REVIEW

File: kolibri/plugins/user/assets/src/views/sign-in-page/index.vue [hunk 1]
```diff
@@ -160,18 +160,21 @@
...
-          this.usernameSuggestions = [];
-        } else if ((!newVal.startsWith(this.suggestionTerm) && this.suggestionTerm.length) ||
-          !this.suggestionTerm.length) {
-          // We have already set a suggestion search term
-          // The currently set suggestion term does not match the current username
-          // Or we do not currently have a suggestion term set
-          // Set it to the new term and fetch new suggestions
-          this.suggestionTerm = newVal;
-          this.setSuggestions();
+        if (newVal) {
```
Comment (indirectlylit) [inline kolibri/plugins/user/assets/src/views/sign-in-page/index.vue]: not familiar with the code, does this need an `else` clause to clear suggestions if they clear the username?

Comment (rtibbles) [reply kolibri/plugins/user/assets/src/views/sign-in-page/index.vue]: A cleared username should be `''` not `null` or `undefined`, so this shouldn't catch it.

Comment (rtibbles) [reply kolibri/plugins/user/assets/src/views/sign-in-page/index.vue]: Woops, should do better null/undefined checking.

**Comment by indirectlylit** [comment]
boo

**Pull request merged**